recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "gcc", "config", "i386", "t-linux64"),
+            "m64=../lib64",
+            "m64=../lib",
+            strict=False,
+        )
+
         # Ensure correct install names when linking against libgcc_s;
         # see discussion in https://github.com/Homebrew/legacy-homebrew/pull/34303
-        tools.replace_in_file(os.path.join(self.source_folder,
-                                           self._source_subfolder, "libgcc", "config", "t-slibgcc-darwin"),
-                              "@shlib_slibdir@", libdir, strict=False)
-        autotools = self._configure_autotools()
-        autotools.make(args=self._make_args)
-
-    def package_id(self):
-        del self.info.settings.compiler
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "libgcc", "config", "t-slibgcc-darwin"),
+            "@shlib_slibdir@",
+            os.path.join(self.package_folder, "lib"),
+            strict=False,
+        )
+
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
-        if self.settings.build_type == "Debug":
-            autotools.install(args=self._make_args)
-        else:
-            autotools.make(args=["install-strip"] + self._make_args)
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.remove_files_by_mask(self.package_folder, "*.la")
-        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+        autotools = Autotools(self)
+        autotools.make(target="install-strip")
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", self.package_folder, recursive=True)
+        copy(
+            self,
+            pattern="COPYING*",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+            keep_path=False,
+        )
 
     def package_info(self):
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : " + bindir)
-        self.env_info.PATH.append(bindir)
+        if self.info.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.append("m")
+            self.cpp_info.system_libs.append("rt")
+            self.cpp_info.system_libs.append("pthread")
+            self.cpp_info.system_libs.append("dl")
 
-        cc = os.path.join(bindir, "gcc-%s" % self.version)
-        self.output.info("Creating CC env var with : " + cc)
-        self.env_info.CC = cc
+        bindir = os.path.join(self.package_folder, "bin")
 
-        cxx = os.path.join(bindir, "g++-%s" % self.version)
-        self.output.info("Creating CXX env var with : " + cxx)
-        self.env_info.CXX = cxx
+        cc = os.path.join(bindir, f"gcc-{self.version}")
```

Comment:
does anybody have an opinion on this naming convention? This is the convention that existed in the previous version of the recipe, however it's possible that a consumer might expect that given that the `bin` folder is placed on the `PATH` in any `Virtual{Build,Run}Env`, `gcc` would resolve to `self.package_folder/bin/gcc`. That won't happen here because it will be named `gcc-X.Y.Z`. I think I'd be in favor of leaving the executables named their name, i.e. `gcc` in this case. Anybody have any thoughts about what would be best?

Reply:
Let's not change it -- it's safer to explicitly need to set the same version.
If there was another GCC installed and it picked the wrong one this would be difficult.

I agree is more clunky but install gcc-7.2 call gcc-7.2 make sense in my head -- people can add their own alias is they like?